src/main/java/dev/aziz/movies/controllers/MovieController.java

```diff
@@ -0,0 +1,33 @@
+package dev.aziz.movies.controllers;
+
+import dev.aziz.movies.dtos.MovieDto;
+import dev.aziz.movies.dtos.ReducedMovieDto;
+import dev.aziz.movies.services.MovieService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
+@RestController
+@RequiredArgsConstructor
+public class MovieController {
+
+    private final MovieService movieService;
+
+    @GetMapping("/movies")
+    public List<ReducedMovieDto> getAllMovies() {
+        List<MovieDto> movies = movieService.getMovies();
+        List<ReducedMovieDto> reducedMovieDto = movies.stream()
```

Comment:
The mapping should be done in the service. The controller is only responsible to call services, nothing more.